### PR TITLE
Reworked add expense workflow

### PR DESCRIPTION
**Pull request by shashank-madan**

### CODE REVIEW

File: code/add.py [hunk 1]
```diff
@@ -18,26 +18,29 @@ def run(message, bot):
...
+#     markup.row_width = 2
+#     chat_id = message.chat.id
+#     if str(message.text) == "Y" or str(message.text) == "y":
+#         message1 = bot.send_message(chat_id, "Please enter your category")
+#         bot.register_next_step_handler(message1, post_append_spend, bot)
+#     else:
+#         for c in helper.getSpendCategories():
+#             markup.add(c)
+#         msg = bot.reply_to(message, "Select Category", reply_markup=markup)
+#         bot.register_next_step_handler(msg, post_category_selection, bot)
```
Comment (srina1h) [inline code/add.py:43]: Are these comments necessary?

File: code/add.py [hunk 2]
```diff
@@ -67,24 +70,28 @@ def post_category_selection(message, bot):
...
+        if selected_category == "Add new category":
+            message1 = bot.send_message(chat_id, "Please enter your category")
+            bot.register_next_step_handler(message1, post_append_spend, bot)
+        else:
+            if selected_category not in helper.getSpendCategories():
+                bot.send_message(
+                    chat_id, "Invalid", reply_markup=types.ReplyKeyboardRemove()
+                )
+                raise Exception(
+                    'Sorry I don\'t recognise this category "{}"!'.format(selected_category)
```
Comment (srina1h) [inline code/add.py:82]:
```suggestion
                    'Sorry, I don\'t recognise this category "{}"!'.format(selected_category)
```

File: code/add.py [hunk 3]
```diff
@@ -67,24 +70,28 @@ def post_category_selection(message, bot):
...
+                )
+
+            option[chat_id] = selected_category
+            message = bot.send_message(
+                chat_id,
+                "How much did you spend on {}? \n(Enter numeric values only)".format(
```
Comment (srina1h) [inline code/add.py:88]:
```suggestion
                "How much did you spend on {}? \n(Numeric values only)".format(
```

**Review by srina1h — CHANGES_REQUESTED** [review]

File: code/add.py (hunk 1 above)
Comment (shashank-madan) [reply code/add.py]: Thanks for catching this. Have removed them

File: code/add.py (hunk 3 above)
Comment (shashank-madan) [reply code/add.py]: Changed as suggested

File: code/add.py (hunk 2 above)
Comment (shashank-madan) [reply code/add.py]: Changed as suggested

**Review by srina1h — APPROVED** [review]

**Pull request merged**